dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -1715,6 +1715,52 @@ public void executeProcess(ProcessBuilder pb, File logFile) {
     }
   }
 
+  protected  synchronized void recordSnappyProcessIDinNukeRun(String hostName, String pName){
+    String cmd = "ssh -n -x -o PasswordAuthentication=no -o StrictHostKeyChecking=no " + hostName;
+    Process pr = null;
+    try {
+      String dest = new File(".").getCanonicalPath() + File.separator + "PIDs_" + pName + "_" +
+          hostName + ".log";
+      File logFile = new File(dest);
+      if (!logFile.exists()) {
+        cmd += " jps | grep " + pName + " | awk '{print $1}'";
+        hd = TestConfig.getInstance().getMasterDescription()
+            .getVmDescription().getHostDescription();
+        ProcessBuilder pb = new ProcessBuilder("/bin/bash", "-c", cmd);
+        pb.redirectOutput(ProcessBuilder.Redirect.appendTo(logFile));
+        pr = pb.start();
+        pr.waitFor();
+        FileInputStream fis = new FileInputStream(logFile);
+        BufferedReader br = new BufferedReader(new InputStreamReader(fis));
+        String str = null;
+        while ((str = br.readLine()) != null) {
+          int pid = Integer.parseInt(str);
+          try {
+            if (pids.contains(pid)) {
+              Log.getLogWriter().info("Pid is already recorded with Master" + pid);
+            } else {
+              pids.add(pid);
+              RemoteTestModule.Master.recordPID(hd, pid);
+              SnappyBB.getBB().getSharedMap().put("pid" + "_" + pName + "_" + str, str);
+              SnappyBB.getBB().getSharedMap().put("host" + "_" + pid + "_" + hostName, hostName);
+            }
+          } catch (RemoteException e) {
+            String s = "Unable to access master to record PID: " + pid;
+            throw new HydraRuntimeException(s, e);
+          }
+          Log.getLogWriter().info("pid value successfully recorded with Master");
+        }
+        br.close();
+      }
+    } catch (IOException e) {
+      String s = "Problem while starting the process : " + pr;
+      throw new TestException(s, e);
+    } catch (InterruptedException e) {
+      String s = "Exception occurred while waiting for the process execution : " + pr;
+      throw new TestException(s, e);
+    }
+  }
+
   protected synchronized void recordSnappyProcessIDinNukeRun(String pName) {
```

Comment:
I think, this method cab be removed now.

Reply:
This is still being used by the previous methods, so will keep it.